geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/consumer/ThreadedConsumer.scala

```diff
@@ -110,3 +114,21 @@ abstract class ThreadedConsumer(
     }
   }
 }
+
+object ThreadedConsumer {
+
+  /**
+    * Handler for asynchronous errors in the consumer threads
+    */
+  trait ConsumerErrorHandler {
+
+    /**
+      * Invoked on a fatal error
+      *
+      * @param consumer consumer identifier
+      * @param e exception
+      * @return true to continue processing, false to terminate the consumer
+      */
+    def handle(consumer: String, e: Throwable): Boolean
```

Comment:
Ok, this is cool.  How does one configure one?:)

Reply:
you pass it off to the `startConsumers` method here: https://github.com/locationtech/geomesa/pull/2333/files#diff-7f9d3345259eca082a59f009c3f485a5R52

Reply:
Ok, so one would have to do this quite manually?

Reply:
Yeah, the idea is to give access to the extending class, as currently everything is private and inaccessible, but without exposing too much inner logic.